src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -291,23 +291,42 @@ private static function colorize($str, $red)
      * @param string[]      $groups
      * @param Configuration $configuration
      * @param bool          $isFailing
+     *
+     * @throws InvalidArgumentException
      */
     private function displayDeprecations($groups, $configuration, $isFailing)
     {
         $cmp = function ($a, $b) {
             return $b->count() - $a->count();
         };
 
+        if ($configuration->shouldWriteToLogFile()) {
+            if (false === $handle = @fopen($file = $configuration->getLogFile(), 'a')) {
+                throw new \InvalidArgumentException(sprintf('The configured log file "%s" is not writeable.', $file));
+            }
+        } else {
+            $handle = \STDOUT;
+        }
+
         foreach ($groups as $group) {
             if ($this->deprecationGroups[$group]->count()) {
-                echo "\n", self::colorize(
-                    sprintf(
-                        '%s deprecation notices (%d)',
-                        \in_array($group, ['direct', 'indirect', 'self'], true) ? "Remaining $group" : ucfirst($group),
-                        $this->deprecationGroups[$group]->count()
-                    ),
-                    'legacy' !== $group && 'indirect' !== $group
-                ), "\n";
+                $deprecationGroupMessage = sprintf(
+                    '%s deprecation notices (%d)',
+                    \in_array($group, ['direct', 'indirect', 'self'], true) ? "Remaining $group" : ucfirst($group),
+                    $this->deprecationGroups[$group]->count()
+                );
+                if ($configuration->shouldWriteToLogFile()) {
+                    echo "\n", self::colorize(
```

Comment:
on one line

also can't the call be moved outside of the if/else?
or the if/else be removed altogether?

Reply:
True.. I refactored it, thank you!